nixos/modules/system/boot/systemd.nix

```
@@ -673,6 +675,28 @@ in
       '';
     };
 
+    services.sleep.hibernateDelaySec = mkOption {
+      default = 3600;
+      type = types.int;
```

```suggestion
      type = with types; nullOr int;
```
and `default = null`, and appropriate config to NOT enable hibernate when `hibernateDelaySec == null` . It is a bad idea to enable suspend-then-hibernate for all, because not everyone has enough swap to manage hibernate.

---

I don't know if we really need this. As I understand [the docs](suspend-then-hibernate) you must call suspend-then-hibernate explicitly to get the hibernation. Suspend will still just supend and never got to hibernate.